tests/circuit_qsim_parser_test.cc

```diff
@@ -70,6 +70,24 @@ R"(2
   EXPECT_EQ(circuit.gates.size(), 10);
 }
 
+TEST(CircuitQsimParserTest, ValidCircuitWithControlledGates) {
+  constexpr char valid_circuit[] =
+R"(5
+0 c 0 1 h 2
```

Comment:
I don't think this is a valid circuit. A qubit can't simultaneously control one gate and be acted on by another, as is the case with qubits {0, 1, 2} in this example.

Reply:
Done.

Reply:
The fact that the test previously passed suggests that this is still supported in the parser. Could you update the parser and add a test to make sure that simultaneous control and action on a qubit is rejected?

Reply:
Simultaneous control and action on a qubit is rejected, see lines 381 and 393 in this files. You mean that it is possible to have gates with equal times (or moments in Cirq language) that act on the same qubits, right? This is true for all the gates not only for controlled gates. So perhaps that should be fixed by a separate PR.

Reply:
Ah, I wasn't aware that was the case. In that case it's fine to leave this to a separate PR - I'll open an issue to track it.